### DISCUSSION TITLE
type and defaultValue with getInputProps

### DISCUSSION THREAD

**Discussion by ghost** [airjp73/rvf#61] [Q&A] [answered]
Hey,

I have a validation like this: `amount: zfd.numeric(z.number().default(0)),`

At first I was expecting type and defaultValue to be set with `...getInputProps`, but it doesn't.
Wouldn't that be a nice feature?

I now created an Input component like below, but `defaultValue` doesn't work.
Do I use this wrong?

```javascript
type MyInputProps = { name: string; label: string; type: string };

export const MyInput = ({ name, label, type }: MyInputProps) => {
  const { error, getInputProps, defaultValue } = useField(name);
  return (
    <div>
      <label htmlFor={name}>{label}</label>
      <input
        {...getInputProps({ id: name, type })}
        defaultValue={defaultValue}
      />
      {error && <span className="my-error-class">{error}</span>}
    </div>
  );
};
```

**Comment by airjp73** [comment, marked as answer]
Hi!

We don't pull default values out of the validation schema. Defaults built into the schema itself should be used to specify what value you want if the user leaves that input empty.

You can provide default values for your form in one of two ways:

* The `defaultValues` prop on the `ValidatedForm`
  * https://www.remix-validated-form.io/default-values
* Returning `setFormDefaults` from your loader
  * https://www.remix-validated-form.io/reference/set-form-defaults

If you do this, you don't actually need to handle the `defaultValue` from `useField` at all and it should still work correctly.

```diff
type MyInputProps = { name: string; label: string; type: string };

export const MyInput = ({ name, label, type }: MyInputProps) => {
- const { error, getInputProps, defaultValue } = useField(name);
+ const { error, getInputProps } = useField(name);
  return (
    <div>
      <label htmlFor={name}>{label}</label>
      <input
        {...getInputProps({ id: name, type })}
-       defaultValue={defaultValue}
      />
      {error && <span className="my-error-class">{error}</span>}
    </div>
  );
};
```

**Reply by ghost** [reply]
Thanks! I should have seen that in the docs.
I only made the `type` optional now. 

```typescript
import { useField } from "remix-validated-form";

type MyInputProps = { name: string; label: string; type?: string };

export const MyInput = ({ name, label, type }: MyInputProps) => {
  const { error, getInputProps } = useField(name);
  type ||= "text";
  return (
    <div>
      <label htmlFor={name}>{label}</label>
      <input {...getInputProps({ id: name, type })} />
      {error && <span className="my-error-class">{error}</span>}
    </div>
  );
};
```
